framework/src/main/java/org/checkerframework/common/wholeprograminference/SceneToStubWriter.java

```diff
@@ -0,0 +1,605 @@
+                // (see AnnotationBuilder#addEmptyArrayField)
+                if (aaft.elementType == null) {
+                    if (l.size() != 0) {
+                        throw new AssertionError();
+                    }
+                } else {
+                    StringJoiner sj = new StringJoiner(",");
+                    for (Object o2 : l) {
+                        sj.add(formatAnnotationValue(aaft.elementType, o2));
+                    }
+                    result.append(sj.toString());
+                }
+            }
+            result.append("}");
+            return result.toString();
+        } else if (aft instanceof ClassTokenAFT) {
+            return aft.format(o);
+        } else if (aft instanceof BasicAFT && o instanceof String) {
+            return Strings.escape((String) o);
+        } else if (aft instanceof BasicAFT && o instanceof Long) {
+            return o.toString() + "L";
+        } else {
+            return o.toString();
+        }
+    }
+
+    /**
+     * Returns the String representation of an annotation in Java source format.
+     *
+     * @param a the annotation to print
+     * @return the formatted annotation
+     */
+    private String formatAnnotation(Annotation a) {
+        StringBuilder result = new StringBuilder();
+        result.append("@" + a.def().name.substring(a.def().name.lastIndexOf('.') + 1));
+        if (!a.fieldValues.isEmpty()) {
+            result.append('(');
+            StringJoiner sj = new StringJoiner(",");
+            for (Map.Entry<String, Object> f : a.fieldValues.entrySet()) {
+                AnnotationFieldType aft = a.def().fieldTypes.get(f.getKey());
+                sj.add(f.getKey() + "=" + formatAnnotationValue(aft, f.getValue()));
+            }
+            result.append(sj.toString());
+            result.append(')');
+        }
+        return result.toString();
+    }
+
+    /**
+     * Returns all annotations in {@code annos} in a form suitable to be printed as Java source
+     * code.
+     *
+     * <p>Each annotation is followed by a space, to separate it from following Java code.
+     *
+     * <p>Internal JDK annotations, such as jdk.Profile+Annotation, are ignored. These annotations
+     * are identified using the convention that their names contain "+".
+     *
+     * @param annos the annotations to format
+     * @return all annotations in {@code annos}, separated by spaces, in a form suitable to be
+     *     printed as Java source code
+     */
+    private String formatAnnotations(Collection<? extends Annotation> annos) {
+        StringBuilder sb = new StringBuilder();
+        for (Annotation tla : annos) {
+            if (!tla.def.name.contains("+")) {
+                sb.append(formatAnnotation(tla));
+                sb.append(" ");
+            }
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Formats the annotations on the component type of an array, if there are any.
```

Comment:
This says "formats the annotations", but the `@return` clause says it returns the array type, not just the annotations.